corehq/apps/locations/expressions.py

```diff
@@ -0,0 +1,20 @@
+from corehq.apps.userreports.specs import TypeProperty
+from dimagi.ext.jsonobject import JsonObject
+
+
+class LocationTypeSepc(JsonObject):
+    type = TypeProperty('location_type_name')
+
+    def __call__(self, item, context=None):
+        try:
+            doc_id = item['_id']
+            from corehq.apps.locations.models import SQLLocation
+            sql_location = SQLLocation.objects.filter(location_id=doc_id)
```

Comment:
there are a couple cases that could cause it. the main one is that if you have multiple columns that depend on the location type it will get called once for each of those for the same document.

however, if you submit multiple forms back to back with the same location type - and are using this expression in a form or case table - then it would also be cached between form submissions.